src/wrapper/macros.rs

```diff
@@ -25,7 +25,7 @@ macro_rules! jni_call_check_ex {
     ( $jnienv:expr, $version:tt, $name:tt $(, $args:expr )* ) => ({
         let ret = jni_call_unchecked!($jnienv, $version, $name $(, $args)*);
         if $jnienv.exception_check() {
-            Err(crate::errors::Error::JavaException)
```

Comment:
I'm not confident this is the way to go...

It feels a bit funky to be automatically clearing + re-throwing all exceptions for the sake of creating a global ref that may or may not be used afterwards.

Should we instead be providing utilities that make it ergonomic to capture an exception explicitly, but leave it up to `jni` users to decide if they want to do that from Rust or else let them decide to return back to the JVM so it will be thrown within the JVM instead?

For example in the android-activity crate I have a utility that maps a `JavaException` into an Error: https://github.com/rust-mobile/android-activity/blob/967882f3d904d9bdc660dee96855c874a8968ce3/android-activity/src/jni_utils.rs#L57

Reply:
As I recall, the reason I did this was to guarantee that the *correct* exception is displayed. As far as I know, the only way to do that is to catch it immediately when it's thrown, and retain a reference to it in case it is displayed later.

If the exception is caught only when it's to be displayed, then it's possible that the exception was already caught, cleared, and some *other* exception (or none at all) is now pending instead.

But yeah, I was wondering (in the original PR comment) whether such a guarantee is worth the performance overhead and code complexity. I take it you're thinking not?